src/musicbrainz/musicbrainzxml.cpp

```diff
@@ -393,6 +393,7 @@ QPair<QList<TrackRelease>, bool> readElementRecording(QXmlStreamReader& reader)
                 while (continueReading(reader)) {
                     const QXmlStreamReader::TokenType type = reader.readNext();
                     if (type == QXmlStreamReader::StartElement) {
+                        const QStringRef name = reader.name();
```

Comment:
You are defining a variable with the same name as a different variable in the surrounding scope. This is a recipie for very hard to catch bugs like #1447.

Reply:
I am not going to touch this code again. Manual parsing is a pita.

Reply:
The only alternative is to replace it with the call `reader.name()` literally.

Reply:
I am not going to split the code into separate functions.

Reply:
I am not suggesting splitting the code into separate functions. But you can rename one of the variables.

Reply:
Giving them arbitrary names is not an option for me.

Reply:
`readerName` wouldn't be arbitrary IMHO.

Reply:
How about making the outer variable `startElementName` and the inner one `releaseListName`?

Reply:
I just fixed the bug. Everything else would add unnecessary noise to this PR. We are expecting one or more values, but we don't know in advance what's in there. The proposed names don't fit.

The variable is just an alias of what's currently stored in reader.name(). Using `readerName` doesn't twice doesn't make it better. I will not change anything here unless it is a real improvement.

Reply:
This is a highly error prone solution. This is not hypothetical. I remember it took me 8 hours to solve #1447. Even if the variable names are not that descriptive, IMO we should never in any circumstances define a variable with the same name as the surrounding scope. It is surprising that compilers even allow this.

Reply:
You could even call it `nameInner` for lack of a better term if you want. Just not the same name as the outer scope.

Reply:
Now it is consistent with the remaining code, i.e. no variables at all.

Reply:
Sure, that works too.